app/src/main/java/de/danoeh/antennapod/fragment/preferences/SynchronizationProviderViewData.java

```diff
@@ -0,0 +1,85 @@
+package de.danoeh.antennapod.fragment.preferences;
+
+import android.content.Context;
+import android.util.Log;
+
+import com.nextcloud.android.sso.exceptions.NextcloudFilesAppAccountNotFoundException;
+import com.nextcloud.android.sso.exceptions.NoCurrentAccountSelectedException;
+import com.nextcloud.android.sso.helper.SingleAccountHelper;
+
+import java.lang.reflect.Field;
+import java.util.LinkedHashMap;
+
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.core.preferences.GpodnetPreferences;
+
+import static de.danoeh.antennapod.core.sync.SyncService.SYNC_PROVIDER_CHOICE_GPODDER_NET;
+import static de.danoeh.antennapod.core.sync.SyncService.SYNC_PROVIDER_CHOICE_NEXTCLOUD;
+
+public class SynchronizationProviderViewData {
+
+    public static String getUsernameFromSelectedSyncProvider(Context context, String currentSyncProviderKey) {
+        switch (currentSyncProviderKey) {
+            case SYNC_PROVIDER_CHOICE_GPODDER_NET:
+                return GpodnetPreferences.getUsername();
+            case SYNC_PROVIDER_CHOICE_NEXTCLOUD:
+                try {
+                    return SingleAccountHelper.getCurrentSingleSignOnAccount(context).name;
+                } catch (NextcloudFilesAppAccountNotFoundException e) {
+                    e.printStackTrace();
+                    return "";
+                } catch (NoCurrentAccountSelectedException e) {
+                    e.printStackTrace();
+                    return "";
+                }
+            default:
+                return "";
+        }
+
+    }
+
+    public static int getSynchronizationProviderHeaderSummary(Context context, String provider) {
+        return getSynchronizationProviderPropertyId(context, provider, R.array.sync_provider_summary, R.string.class);
+    }
+
+    public static int getSynchronizationProviderIcon(Context context, String provider) {
+        return getSynchronizationProviderPropertyId(context, provider, R.array.sync_provider_icon, R.drawable.class);
+    }
+
+    private static int getSynchronizationProviderPropertyId(
+            Context context,
+            String provider,
+            int propertyArrayId,
+            Class type
+    ) {
+        final String[] availableProviders = context.getResources().getStringArray(R.array.sync_provider_keys);
+        final LinkedHashMap<String, String> syncProvidersValues = getSyncProviderValuesByProperty(
+                context,
+                availableProviders,
+                propertyArrayId
+        );
+        String resourceKey = syncProvidersValues.get(provider);
+        int stringId = 0;
+        try {
+            Field field = type.getField(resourceKey);
```

Comment:
Hmmm. Please avoid using reflection. It took me quite long to understand what this does. When using proguard with variable name compression (which we currently don't), this could also break. How about making an `enum` that has 3 instances (none, gpodder, nextcloud) and each instance has attributes for name, icon, etc?

Reply:
thanks. i did not know about that construct. i'll try

Reply:
Awesome. That makes it way more easy. Thanks for the hint